src/main/java/org/jitsi/jigasi/xmpp/CallControlMucActivator.java

```diff
@@ -0,0 +1,422 @@
+import net.java.sip.communicator.service.protocol.jabber.*;
+import net.java.sip.communicator.util.*;
+import net.java.sip.communicator.util.Logger;
+import org.jitsi.jigasi.*;
+import org.jitsi.jigasi.util.*;
+import org.jitsi.service.configuration.*;
+
+import org.jivesoftware.smack.*;
+import org.jivesoftware.smack.filter.*;
+import org.jivesoftware.smack.packet.*;
+import org.osgi.framework.*;
+
+import java.util.*;
+
+/**
+ * @author Damian Minkov
+ */
+public class CallControlMucActivator
+    implements BundleActivator,
+               ServiceListener,
+               RegistrationStateChangeListener,
+               GatewaySessionListener,
+               SipGatewayListener,
+               PacketListener,
+               PacketFilter
+{
+    /**
+     * The logger
+     */
+    private final static Logger logger
+        = Logger.getLogger(CallControlMucActivator.class);
+
+    public static BundleContext osgiContext;
+
+    /**
+     * The account property to search in configuration service for the room name
+     * where all xmpp providers will join.
+     */
+    private static final String ROOM_NAME_ACCOUNT_PROP = "BREWERY";
+
+    /**
+     * A property to enable or disable muc call control, disabled by default.
+     */
+    private static final String BREWERY_ENABLED_PROP
+        = "org.jitsi.jigasi.BREWERY_ENABLED";
+
+    /**
+     * The call controlling logic.
+     */
+    private CallControl callControl = null;
+
+    /**
+     * Starts muc control component. Finds all xmpp accounts and listen for
+     * new ones registered.
+     * @param bundleContext
+     * @throws Exception
+     */
+    @Override
+    public void start(final BundleContext bundleContext)
+        throws Exception
+    {
+        osgiContext = bundleContext;
+
+        ConfigurationService config = getConfigurationService();
+
+        if (!config.getBoolean(BREWERY_ENABLED_PROP, false))
+        {
+            logger.warn("MUC call control disabled.");
+            return;
+        }
+
+        osgiContext.addServiceListener(this);
+
+        Collection<ServiceReference<ProtocolProviderService>> refs
+            = ServiceUtils.getServiceReferences(
+            osgiContext,
+            ProtocolProviderService.class);
+
+        for (ServiceReference<ProtocolProviderService> ref : refs)
+        {
+            initializeNewProvider(osgiContext.getService(ref));
+        }
+
+        SipGateway gateway = ServiceUtils.getService(
+            bundleContext, SipGateway.class);
+        if (gateway != null)
+        {
+            gateway.addSipGatewayListener(this);
+
+            if (this.callControl == null)
+                this.callControl = new CallControl(gateway, config);
+        }
+    }
+
+    /**
+     * Stops and unregister everything - providers, listeners.
+     * @param bundleContext
+     * @throws Exception
+     */
+    @Override
+    public void stop(BundleContext bundleContext)
```

Comment:
this.callControl reference is not cleared. Also does it require to release any resources ?